src/chigame/games/models.py

```diff
@@ -34,13 +36,39 @@ class Game(models.Model):
     min_playtime = models.PositiveIntegerField(null=True, blank=True)
     max_playtime = models.PositiveIntegerField(null=True, blank=True)
 
-    complexity = models.PositiveSmallIntegerField(null=True, blank=True)  # 1-5, 1 being the easiest
-    category = models.ManyToManyField("Category", related_name="games", blank=True)
+    complexity = models.DecimalField(
+        max_digits=3, decimal_places=2, null=True, validators=[MinValueValidator(1), MaxValueValidator(5)]
+    )
+
+    categories = models.ManyToManyField("Category", related_name="games", blank=True)
     mechanics = models.ManyToManyField("Mechanic", related_name="games", blank=True)
 
     # ================ OTHER ================
     BGG_id = models.PositiveIntegerField(null=True, blank=True)  # BoardGameGeek ID
 
+    # ================ VALIDATON ================
+    def clean(self):
+        # Ensures min_players is not greater than max_players
+        if self.min_players and self.max_players and self.min_players > self.max_players:
+            raise ValidationError({"min_players": "min_players cannot be greater than max_players"})
+
+        # Validate playtime constraints for all combinations of min_playtime, max_playtime, and expected_playtime
+        if self.min_playtime is not None and self.max_playtime is not None:
+            if self.min_playtime > self.max_playtime:
+                raise ValidationError({"min_playtime": "min_playtime cannot be greater than max_playtime"})
+
+        if self.expected_playtime is not None:
+            if self.min_playtime is not None and self.expected_playtime < self.min_playtime:
+                raise ValidationError({"expected_playtime": "expected_playtime cannot be less than min_playtime"})
+
+            if self.max_playtime is not None and self.expected_playtime > self.max_playtime:
+                raise ValidationError({"expected_playtime": "expected_playtime cannot be greater than max_playtime"})
+
+    def save(self, *args, **kwargs):
+        # Calls full_clean to run all validations before saving
```

Comment:
Given that this calls `full_clean` whereas you just defined the function `clean` above, consider adding a comment which clarifies how/that `full_clean` runs `clean`.
A link to documentation would be helpful since this feature is not covered in class notes!

Reply:
Comment added!